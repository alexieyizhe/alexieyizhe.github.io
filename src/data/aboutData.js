import React from "react";
import Link from "gatsby-link";

export const STATS_COUNTER_DURATION = 3;

export const STATS_LIST = [
  {
    name: 'Trips Around The Sun',
    start: 0,
    end: 18,
  },
  {
    name: 'Hours Spent Coding',
    start: 0,
    end: 2984,
  },
  {
    name: 'Soccer Balls Kicked',
    start: 0,
    end: 973,
  },
  {
    name: 'Unpronounceable Last Name',
    start: 0,
    end: 1,
  }
];

export const DESC_PARAGRAPHS = [
  <span>
    I'm a diehard soccer fan, whether it's watching FC Barcelona - my favourite team - or getting
    on the field myself. Apart from soccer and coding, my other interests are <s>nonexistent</s> fitness,
    cooking, my husky-malamute Storm, and travelling. I've done a couple of solo trips that you can read about
    on <Link to="/blog">my blog</Link>!
  </span>,
  <span>
    My friends always tell me that I'm addicted to coding, but I like to think of it less as an addiction to coding
    and more as a passion for solving problems. I find myself engrossed in creating solutions to tough problems
    and pushing myself to always improve my skills and abilities; before you know it, I've been programming for
    hours upon hours. This is also one aspect of my skills that I'm constantly improving, so I'm also interested
    in more sustainable and long-term coding practices like Agile development and working in a professional environment.
  </span>,
  <span>
    However, I also believe that creating solutions to problems and learning new ideas aren't limited to the
    scope of studying and coding. I'm currently working with a huge amount of passionate individuals on organizing
    a <a href="https://www.tedxuw.com/" target="_blank">700+ attendee TEDx event</a>, powering one of
    the <a href="https://teamwaterloop.ca/" target="_blank">world's top 25 hyperloop teams</a>, and leading the next iteration
    of <a href="https://equithon.org/" target="_blank">Waterloo's largest social innovation hackathon</a>. These have been
    incredibly rewarding learning experiences that I cherish greatly, and I'm hoping to continue contributing and getting
    involved with various communities now and into the future.
  </span>,
  <span>
    At my previous co-op at Flipp, I worked with some other awesome peeps on the system
    responsible for indexing retailer products to power the results on Flipp's flyers and search results.
    Even though almost everything I worked on at Flipp is on the back-end, I'm extremely interested
    in the workings of the entire web stack. Stemming from the fact that I'm a very visual learner, I
    have a deep fascination with user interfaces and UX design. I absolutely love trawling the web for interesting
    articles and demos/case studies of said topic - stuff like human-computer interactions and subtle effects on
    user experience are my bread and butter. It's a major contributing factor to why I'm learning JavaScript,
    React, responsive design, and how to use tools like Figma and Adobe XD.
  </span>,
  <span>
    If you've made it this far, props (no React pun intended) to you ðŸ¥‚ I'm always looking for new initiatives.
    If you have any questions or wanna chat, shoot me an email or find me on social media!
  </span>
]


export const INTERESTS_LIST = [
  { name: "Travel", icon: "briefcase" },
  { name: "Fitness", icon: "activity" },
  { name: "Games", icon: "crosshair" },
  { name: "Coding", icon: "code" },
  { name: "Volunteering", icon: "heart" }
]

export const SKILLS_LIST = [
  {
    type: 'languages',
    children: [
      { name: "JavaScript", icon: "javascript" },
      { name: "Ruby", icon: "ruby" },
      { name: "Python", icon: "python" },
    ]
  },
  {
    type: 'frameworks',
    children: [
      { name: "React", icon: "atom" },
      { name: "Ruby on Rails", icon: "ruby_on_rails" },
      { name: "HTML", icon: "html5" },
      { name: "CSS", icon: "css3" },
      { name: "Ionic 3", icon: "ionic" },
      { name: "Angular 4", icon: "angular" },
      { name: "MeteorJS", icon: "meteor" },
    ]
  },
  {
    type: 'tools',
    children: [
      { name: "SQL Databases", icon: "database" },
      { name: "Terminal (Bash)", icon: "terminal" },
      { name: "MongoDB & NoSQL DBs", icon: "mongodb" },
      { name: "Git", icon: "git_branch" },
      { name: "Creative Problem Solving", icon: "stackoverflow" },
    ],
  }
]
