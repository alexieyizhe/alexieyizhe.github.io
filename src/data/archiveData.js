export const archiveSiteList = [
  {
    name: 'May 2018',
    url: '/archive/may-2018/',
    desc: 'A snapshot of alexieyizhe.me in May 2018. This iteration was the final one before V2 (the current site you\'re seeing here).',
  },
  {
    name: 'Jan 2018',
    url: '/archive/jan-2018/',
    desc: 'A snapshot of alexieyizhe.me in Jan 2018. This was the initial iteration of my portfolio site without too much polish.',
  }
]
