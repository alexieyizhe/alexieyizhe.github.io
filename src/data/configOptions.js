import { css } from 'styled-components';

const displaySizes = {
  desktop: 2160,
  tablet: 1024,
  phone: 425
}

// Iterate through the sizes and create a media template
export const mediaSize = Object.keys(displaySizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (max-width: ${displaySizes[label] / 16}em) {
      ${css(...args)}
    }
  `
  return acc
}, {})



export const menuPageOptions = [
  {
    text: 'Home',
    route: '/',
    colour: '#F7A3F2'
  },
  {
    text: 'About',
    route: '/about',
    colour: '#8EE8A7'
  },
  {
    text: 'Experience',
    route: '/experience',
    colour: '#F4B276'
  },
  {
    text: 'Projects',
    route: '/projects',
    colour: '#A6CAF5'
  },
  {
    text: 'Blog',
    route: '/blog',
    colour: '#E7EC83'
  },
  {
    text: 'Archive',
    route: '/archive',
    colour: '#EE9AB8'
  },
];

export const contactOptions = [
  {
    text: 'say hi!',
  },
  {
    text: 'resume'
  },
  {
    text: 'github'
  },
  {
    text: 'linkedin'
  }
];

export const resumeOptions = [
  {
    name: 'Long Ago',
    previewSource: '/img/resume/2013-resume.jpg',
    downloadName: 'Alex Xie - Resume (I was 13 when I made this ok)',
    downloadSource: '/docs/2013-resume.pdf',
    color: '#cdc66b'
  },
  {
    name: 'Oct 2017',
    previewSource: '/img/resume/2017-10-resume.png',
    downloadName: 'Alex Xie - Resume (Oct 2017)',
    downloadSource: '/docs/2017-10-resume.pdf',
    color: '#F4B276'
  },
  {
    name: 'Feb 2018',
    previewSource: '/img/resume/2018-02-resume.png',
    downloadName: 'Alex Xie - Resume (Feb 2018)',
    downloadSource: '/docs/2018-02-resume.pdf',
    color: '#8EE8A7'
  },
  {
    name: 'July 2018',
    previewSource: '/img/resume/2018-07-resume.png',
    downloadName: 'Alex Xie - Resume (Aug 2018)',
    downloadSource: '/docs/2018-07-resume.pdf',
    color: '#EE9AB8'
  },
  {
    name: 'Current',
    previewSource: '/img/resume/2019-01-resume.png',
    downloadName: 'Alex Xie - Resume',
    downloadSource: '/docs/2019-01-resume.pdf',
    color: '#88dbe3'
  },
];

export const greetingOptions = ['Hi!', 'Hey!', 'Sup?', 'Howdy!', 'Heya!', 'Yo!', 'ä½ å¥½!', 'Bonjour!', 'ðŸ‘‹ !']

export const particleConfig = {
  "particles": {
    "number": {
      "value": 80,
      "density": {
        "enable": true,
        "value_area": 3500
      }
    },
    "color": {
      "value": ["#BD10E0","#B8E986","#50E3C2","#FFD300","#E86363"]
    },
    "shape": {
      "type": "circle",
      "stroke": {
        "width": 0,
        "color": "#000000"
      },
      "polygon": {
        "nb_sides": 5
      },
      "image": {
        "src": "img/github.svg",
        "width": 100,
        "height": 100
      }
    },
    "opacity": {
      "value": 0.4,
      "random": true,
      "anim": {
        "enable": false,
        "speed": 1,
        "opacity_min": 0.1,
        "sync": false
      }
    },
    "size": {
      "value": 4.008530152163807,
      "random": true,
      "anim": {
        "enable": false,
        "speed": 40,
        "size_min": 0.1,
        "sync": false
      }
    },
    "line_linked": {
      "enable": false,
      "distance": 150,
      "color": "#ffffff",
      "opacity": 0.4,
      "width": 1
    },
    "move": {
      "enable": true,
      "speed": 2,
      "direction": "none",
      "random": false,
      "straight": false,
      "out_mode": "out",
      "bounce": false,
      "attract": {
        "enable": false,
        "rotateX": 600,
        "rotateY": 1200
      }
    }
  },
  "interactivity": {
    "detect_on": "canvas",
    "events": {
      "onhover": {
        "enable": false,
        "mode": "repulse"
      },
      "onclick": {
        "enable": false,
        "mode": "push"
      },
      "resize": true
    },
    "modes": {
      "grab": {
        "distance": 400,
        "line_linked": {
          "opacity": 1
        }
      },
      "bubble": {
        "distance": 400,
        "size": 40,
        "duration": 2,
        "opacity": 8,
        "speed": 3
      },
      "repulse": {
        "distance": 200,
        "duration": 0.4
      },
      "push": {
        "particles_nb": 4
      },
      "remove": {
        "particles_nb": 2
      }
    }
  },
  "retina_detect": true
}
