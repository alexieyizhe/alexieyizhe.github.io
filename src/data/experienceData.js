export const experienceList = [
  {
    name: 'Flipp Corp.',
    role: 'Software Engineer - Item Data Platform',
    color: '#46CEF9',
    bgImgSource: '/img/experience/flipp_bg.jpg',
    logoImgSource: '../img/experience/flipp_logo.png',
    desc: 'I migrated key systems from a monolith structure to a Ruby & Scala microservice architecture, as well as assisted in improving existing services by adding bug fixes, error detection, and schema validation. I\'m also owning the development of a brand new microservice used in production to monitor, compare, and analyze the effectiveness of both legacy and microservice architectures.'
  },
  {
    name: 'Equithon',
    role: 'Junior Co-Chair & Development Lead',
    color: '#4D55C3',
    bgImgSource: '/img/experience/equithon_bg.jpg',
    logoImgSource: '/img/experience/equithon_logo.png',
    desc: 'I\'m responsible for executive decisions as a co-chair, and I\'m working closely with all team leads to improve upon last year\'s Equithon. As well, I\'m directing the team of developers that are handling all technical aspects of the event in 2019, along with contributing significant code working as part of said team.'
  },
  {
    name: 'TEDxUW',
    role: 'Logistics Manager & Developer',
    color: '#D43F35',
    bgImgSource: '/img/experience/tedx_bg.jpg',
    logoImgSource: '/img/experience/tedx_logo.png',
    desc: 'I\'m part of the executive team coordinating the logistics of the 2019 event, with ambitions to push the event to over 700 attendees. As part of my position, I work closely with other teams to flesh out an event breakdown and smoothly integrate the results of their work into a cohesive event. As well, I\'m also the sole developer on the volunteer and event app for the event.'
  },
  {
    name: 'Waterloop',
    role: 'Sponsorship Associate & Developer',
    color: '#F5D816',
    bgImgSource: '/img/experience/wloop_bg.png',
    logoImgSource: '/img/experience/wloop_logo.png',
    desc: 'I\'m working to secure vital sponsorship funds that power the rest of the team. Waterloop is especially reliant on sponsorship to be self-sufficient compared to many other teams that are backed by large organizations or schools. I\'m also working closely with the Web team to rework Waterloop\'s official website.'
  },
  {
    name: 'Equithon',
    role: 'Developer & Operations Coordinator',
    color: '#4D55C3',
    bgImgSource: '/img/experience/equithon_bg.jpg',
    logoImgSource: '/img/experience/equithon_logo.png',
    desc: 'I was in charge of logistics for the 2018 event, and had the task of scaling up the event to accommodate a 400% increase in the number of attendees while working closely with other members of the team. In addition, I developed the official volunteer app for the event, which was used by all volunteers to keep track of the details of each attendee and interact with the database to track statistics and accurate up-to-date event information.'
  },
]
