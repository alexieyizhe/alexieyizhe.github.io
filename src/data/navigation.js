export const menuPageOptions = [
  {
    text: 'Home',
    route: '/',
    colour: '#F7A3F2'
  },
  {
    text: 'About',
    route: '/about',
    colour: '#8EE8A7'
  },
  {
    text: 'Experience',
    route: '/experience',
    colour: '#E4A871'
  },
  {
    text: 'Projects',
    route: '/projects',
    colour: '#A6CAF5'
  },
  {
    text: 'Blog',
    route: '/blog',
    colour: '#E7EC83'
  },
];
