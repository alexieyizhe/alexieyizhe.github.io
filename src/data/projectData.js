
export const projectsList = [ // eslint-disable-line
  {
    name: 'Equithon Main Site',
    desc: 'Main site of the third iteration of Equithon.',
    color: '#9CCDA1',
    imgSource: '/img/projects/equithon-main.png',
    techStack: [
      {name: 'React', icon: 'devicons devicons-atom', color: '#3DAEFF' },
      {name: 'JavaScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Gatsby', icon: 'devicons devicons-opensource', color: '#7C2ACF' },
      {name: 'Webpack', icon: 'devicons devicons-webplatform', color: '#424242' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "https://equithon.org"},
      {name: "View on Github", icon: "github", url: "https://github.com/equithon/site-main"},
    ]
  },
  {
    name: 'Equithon Landing Page',
    desc: 'A landing/splash page for directing visitors before the main site was completed.',
    color: '#9CCDA1',
    imgSource: '/img/projects/equithon-landing.png',
    techStack: [
      {name: 'React', icon: 'devicons devicons-atom', color: '#3DAEFF' },
      {name: 'JavaScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Gatsby', icon: 'devicons devicons-opensource', color: '#7C2ACF' },
      {name: 'Webpack', icon: 'devicons devicons-webplatform', color: '#424242' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "https://old.equithon.org"},
      {name: "View on Github", icon: "github", url: "https://github.com/equithon/site-landing"},
    ]
  },
  {
    name: 'Personal Site',
    desc: 'Error: Maximum recursion depth exceeded. (memory_address: 0xObebebebe)',
    color: '#9CCDA1',
    imgSource: '/img/projects/alexieyizhewebsite.png',
    techStack: [
      {name: 'React', icon: 'devicons devicons-atom', color: '#3DAEFF' },
      {name: 'JavaScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Gatsby', icon: 'devicons devicons-opensource', color: '#7C2ACF' },
      {name: 'Webpack', icon: 'devicons devicons-webplatform', color: '#424242' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "https://alexieyizhe.me"},
      {name: "View on Github", icon: "github", url: "https://github.com/alexieyizhe/alexieyizhe.github.io"},
    ]
  },
  {
    name: 'EQRScanner',
    desc: 'Heads up to all Equithon 2018 volunteers: your job just got a whole lot simpler!',
    color: '#8185DD',
    imgSource: '/img/projects/eqrscanner.png',
    techStack: [
      {name: 'TypeScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Angular', icon: 'devicons devicons-angular', color: '#8E2704' },
      {name: 'Ionic', icon: 'devicons devicons-ionic', color: '#008CD2' },
      {name: 'MongoDB', icon: 'devicons devicons-mongodb', color: '#03A800' },
      {name: 'Meteor', icon: 'devicons devicons-meteor', color: '#DF6034' },
    ],
    actionLinks: [
      {name: "App Store", icon: "shoppingBag", url: "https://itunes.apple.com/US/app/id1380247687"},
      {name: "Play Store", icon: "playCircle", url: "https://play.google.com/store/apps/details?id=ionic.eqrscanner"},
      {name: "View on Github", icon: "github", url: "https://github.com/alexieyizhe/eqrscanner"},
    ]
  },
  {
    name: 'Easy Recipeasy',
    desc: 'The one stop shop for all your "what the heck should I make for dinner?" needs.',
    color: '#E85A3B',
    imgSource: '/img/projects/easy.png',
    techStack: [
      {name: 'Ruby', icon: 'devicons devicons-ruby', color: '#A82B00' },
      {name: 'Rails', icon: 'devicons devicons-ruby_on_rails', color: '#A82B00' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS', icon: 'devicons devicons-css3', color: '#4173D5' },
      {name: 'Bootstrap', icon: 'devicons devicons-bootstrap', color: '#6122A8' },
      {name: 'Edamam API', icon:  'devicons devicons-codepen', color: '#323232' },
      {name: 'MySQL', icon:  'devicons devicons-mysql', color: '#86A1A3' },
      {name: 'Heroku', icon:  'devicons devicons-heroku', color: '#9943A7' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "http://easyrecipeasy.ml/"},
      {name: "View on Github", icon: "github", url: "https://github.com/alexieyizhe/easy-recipeasy"},
    ]
  },
  {
    name: 'WATBus',
    desc: 'Out of data and wondering when the next GRT bus is? Wonder no longer.',
    color: '#436ABB',
    imgSource: '/img/projects/watbus.png',
    techStack: [
      {name: 'Python', icon: 'devicons devicons-python', color: '#EDDD00' },
      {name: 'Flask Microframework', icon: 'devicons devicons-mootools_badge', color: '#4D5788' },
      {name: 'Grand River Transit API', icon: 'devicons devicons-codepen', color: '#323232' },
      {name: 'Heroku', icon: 'devicons devicons-heroku', color: '#9943A7' },
    ],
    actionLinks: [
      {name: "View on Facebook Messenger", icon: "messageSquare", url: "https://m.me/553923414986147"},
      {name: "View on Github", icon: "github", url: "https://github.com/alexieyizhe/wat-bus"},
    ]
  },
  {
    name: 'Kanye Quest',
    desc: 'Kanye West goes on a Kanye Quest to be the Kanye Best and clean up this Kanye Mess.',
    color: '#E4BB0D',
    imgSource: '/img/projects/kanyequest.png',
    techStack: [
      {name: 'Java', icon: 'devicons devicons-java', color: '#E09C03' },
      {name: 'Graphics2d Library', icon: 'devicons devicons-codepen', color: '#323232' },
    ],
    actionLinks: [
      {name: "View on Github", icon: "github", url: "https://github.com/alexieyizhe/HS_PROJECTS/tree/master/kanye-quest"},
    ]
  }
]
