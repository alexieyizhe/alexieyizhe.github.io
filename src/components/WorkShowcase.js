import React from "react";
import styled from "styled-components";
import posed from "react-pose";
import VisibilitySensor from "react-visibility-sensor";
import { css } from 'styled-components';
import { isMobile } from 'react-device-detect';
import { mediaSize } from "../data/configOptions.js";


const Container = styled.div`
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: 75%;
  height: 32vh;
  margin-bottom: 10vh;
  cursor: pointer;
  color: white;

  display: grid;
  padding: 0 5% 0 5%;
  grid-template-columns: 2fr 1fr;
  grid-template-rows: 4fr 2fr;
  grid-template-areas: "title pic"
                       "role role";

  background: url(${props => props.bg}) center/cover;
  filter: grayscale(50%);
  transition: all 0.3s ease-in;

  /* Pseudo-element for shadow on container during focus */
  &:before {
    content: ' ';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;

    box-shadow: 0 10px 50px 0 rgba(0, 0, 0, 0.5);

    opacity: 0;
    transition: opacity 500ms;
  }

  &:after {
    content: ' ';
    position: absolute;
    z-index: 5;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0.6);
    opacity: 1;
    transition: opacity 500ms;
  }

  ${props => props.focused ? css`
    filter: none;
    background: url(${props => props.bg}) center/cover;

    &:before {
      opacity: 1;
    }

    &:after {
      opacity: 0.5;
    }
    ` : null
  }

  ${mediaSize.tablet`
    width: 80%;

    border-radius: 8px;
    &:before, &:after {
      border-radius: 8px;
    }
  `}

  ${mediaSize.phone`
    width: 80%;
    height: 50vh;
    padding: 0 10% 0 10%;
    margin-bottom: 12vh;

    border-radius: 8px;
    &:before, &:after {
      border-radius: 8px;
    }

    grid-template-columns: 2fr 1fr;
    grid-template-rows: auto auto auto;
    grid-template-areas: "title title"
                         "role role"
                         "pic pic";
  `}

`;

const WorkTitle = styled.span`
  font-family: "Raleway", serif;;
  font-size: 4vw;
  font-weight: bolder;
  position: absolute;
  z-index: 6;
  bottom: 0;
  grid-area: title;

  ${mediaSize.tablet`
    font-size: 5vh;
  `}

  ${mediaSize.phone`
    position: relative;
    bottom: auto;
    top: 0.5em;
    font-size: 5vh;
  `}
`;

const WorkLogo = styled.img`
  grid-area: pic;
  justify-self: center;
  position: relative;
  z-index: 6;
  top: -10%;
  right: -18%;
  max-width: 30vw;
  max-height: 20vh;

  filter: grayscale(100%);
  opacity: 0;
  transition: all 0.3s ease-in;

  ${props => props.focused ? css`filter: none; opacity: 1;` : null}

  ${mediaSize.phone`
    top: 18%;
    right: -30%;
    max-width: 25vh;
    max-height: 20vh;
  `}

`

const WorkRole = styled.div`
  grid-area: role;
  position: relative;
  z-index: 6;
  font-family: "Raleway", serif;;
  font-size: 2vw;

  ${mediaSize.tablet`
    font-size: 3vh;
  `}

  ${mediaSize.phone`
    font-size: 7vmin;
  `}
`;

class WorkShowcase extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focused: false,
    };
  }

  handleFocus(focused) {
    this.setState({focused: focused});
  }

  render() {
    return (
      <VisibilitySensor onChange={(isVisible) => this.handleFocus(isMobile && isVisible)}>
        <Container
          focused={this.state.focused}
          bg={this.props.work.bgImgSource}
          onMouseEnter={() => this.handleFocus(true)}
          onMouseLeave={() => this.handleFocus(false)} >

          <WorkTitle>{this.props.work.name}</WorkTitle>
          <WorkRole>{this.props.work.role}</WorkRole>
          <WorkLogo src={this.props.work.logoImgSource} focused={this.state.focused} />

        </Container>
      </VisibilitySensor>
    );
  }
}


export default WorkShowcase;
