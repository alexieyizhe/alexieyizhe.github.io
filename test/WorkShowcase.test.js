import React from 'react';
import WorkShowcase from '../src/components/WorkShowcase';
import { experienceList } from "../src/data/experienceData";

describe('WorkShowcase component', () => {

  it('should render the work showcase', () => {
    const experience = render(
      <WorkShowcase work={experienceList[0]} />
    );

    expect(experience).toMatchSnapshot();
  });
});
