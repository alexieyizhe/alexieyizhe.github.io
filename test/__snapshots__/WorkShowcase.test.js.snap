// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkShowcase component should render the work showcase 1`] = `
.c7 svg {
  width: 3vh;
  height: 3vh;
  stroke: #FFFFFF;
}

.c6 {
  -webkit-animation-name: cBziYB;
  animation-name: cBziYB;
  -webkit-animation-duration: 3s;
  animation-duration: 3s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.c0 {
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: 75%;
  height: auto;
  margin-bottom: 10vh;
  cursor: pointer;
  color: #F6F6F6;
  border-radius: 5px;
  display: grid;
  padding: 0 5% 0 5%;
  grid-template-rows: auto;
  grid-template-areas: "title" "role" "desc" "indicator";
  background: url(/img/experience/flipp_bg.jpg) center/cover;
  -webkit-filter: grayscale(50%);
  filter: grayscale(50%);
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  -webkit-transition: width 0.75s ease,height 0.75s ease;
  transition: width 0.75s ease,height 0.75s ease;
}

.c0:before {
  content: ' ';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  box-shadow: 0 10px 50px 0 rgba(0,0,0,0.6);
  border-radius: 5px;
  opacity: 0;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c0:after {
  content: ' ';
  position: absolute;
  z-index: 5;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 5px;
  background-color: rgba(0,0,0,0.6);
  opacity: 1;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c1 {
  font-family: 'Raleway',Arial,serif;
  font-size: 5vw;
  font-weight: bold;
  position: relative;
  z-index: 6;
  bottom: 0;
  grid-area: title;
  white-space: nowrap;
  padding-top: 1em;
  padding-bottom: 2px;
}

.c3 {
  position: absolute;
  z-index: 6;
  top: -5vh;
  right: -5vw;
  max-width: 40vw;
  max-height: 25vh;
  opacity: 0;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.c2 {
  grid-area: role;
  position: relative;
  z-index: 6;
  font-family: 'Raleway',serif;
  font-size: 2vw;
  padding-bottom: 1.5em;
}

.c4 {
  grid-area: desc;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  padding-bottom: 0;
  line-height: 1.5;
  font-size: 2.5vmin;
  z-index: 6;
  font-family: 'Average','PT Serif',serif;
  max-width: 100%;
  max-height: 0;
  opacity: 0;
  -webkit-transition: max-height 1s,opacity 0.7s,padding 1s;
  transition: max-height 1s,opacity 0.7s,padding 1s;
}

.c5 {
  grid-area: indicator;
  z-index: 6;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
  margin-bottom: 1vw;
}

@media (max-width:64em) {
  .c0 {
    width: 85%;
    border-radius: 8px;
  }

  .c0:before,
  .c0:after {
    border-radius: 8px;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    height: auto;
    width: 80%;
    padding: 0 10% 0 10%;
    margin-bottom: 4em;
    grid-template-rows: auto auto auto;
    grid-template-areas: "title" "role" "desc" "indicator";
  }
}

@media (max-width:64em) {
  .c1 {
    padding-top: 15vmin;
    font-size: 4.5em;
    font-family: "Cabin","Ubuntu",Arial,serif;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }
}

@media (max-width:26.5625em) {
  .c1 {
    font-size: 2.5em;
    position: relative;
    padding-top: 40vw;
  }
}

@media (max-width:64em) {
  .c3 {
    top: -5vw;
    max-width: 30vmin;
    max-height: 25vmin;
  }
}

@media (max-width:26.5625em) {
  .c3 {
    top: -5vw;
    right: -8vw;
    max-width: 43vw;
    max-height: 60vw;
  }
}

@media (max-width:64em) {
  .c2 {
    font-size: 2em;
    padding-bottom: 1.5em;
  }
}

@media (max-width:26.5625em) {
  .c2 {
    font-size: 5vw;
  }
}

@media (max-width:64em) {
  .c4 {
    max-height: 0;
    padding-bottom: 0;
  }
}

@media (max-width:26.5625em) {
  .c4 {
    font-size: 1em;
    max-height: 0;
    padding-bottom: 0;
  }
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Flipp Corp.
  </span>
  <div
    class="c2"
  >
    Software Engineer - Item Data Platform
  </div>
  <img
    class="c3"
    src="../img/experience/flipp_logo.png"
  />
  <div
    class="c4"
  >
    I migrated key systems from a monolith structure to a Ruby & Scala microservice architecture, as well as assisted in improving existing services by adding bug fixes, error detection, and schema validation. I'm also owning the development of a brand new microservice used in production to monitor, compare, and analyze the effectiveness of both legacy and microservice architectures.
  </div>
  <span
    class="c5"
  >
    <div
      class="c6"
      from="-3"
      to="0"
    >
      <span
        class="c7"
        color="#FFFFFF"
      >
        <svg
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 9l6 6 6-6"
          />
        </svg>
      </span>
    </div>
  </span>
</div>
`;
